.github/workflows/helm-chart-ci.yaml

```
@@ -33,6 +33,26 @@ jobs:
       - name: Verify Docs updated
         run: ./helm-docs.sh
 
+  ensure-image-overrides:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3.4.0
+      - name: Check objects for images without overrides
+        run: |
+          set +e
+          set -o pipefail
+          # Look for image: defintions that are not templated. If we find none, exit is not 0 and we invert the error code to get the
+          # test to pass
+          grep -r "image:" "charts/spire"  | grep "templates/" | grep -v 'image: {{ template "' > /tmp/findings
+          res=$?
+          if [ $res -eq 0 ]; then
+            (echo These templates were found to be using statically defined images and not overridable ones. Please fix.
+            cat /tmp/findings) >> "$GITHUB_STEP_SUMMARY"
+            exit 1
+          fi
```

```suggestion
          if [ "$res" -gt 0 ]; then
            (echo These templates were found to be using statically defined images and not overridable ones. Please fix.
            cat /tmp/findings) >> "$GITHUB_STEP_SUMMARY"
            exit $res
          fi
```

I think the if statement has a mistake.

---

The test I think is correct, as described in the comment. The negated grep throws off the look of it a bit.

The rest of the suggestion looks good though.

---

oh. and res will be 0 in that case, so the exit 1 is correct too I think to get the github job to fail.

---

I pulled in the rest.

---

I think you can simplify the grep a little bit:

```
grep -r "image:" charts/spire  --include "*/templates/*" | grep -v 'image: {{ template "'
```

---

Tried it. It doesn't work...
For testing, you can change the last grep to something else and it should show some 'image: xxxx' entries. it fails to do so with the recommendation. Not sure why off hand.